### ISSUE TITLE
No overloads for "__getitem__" match the provided arguments (regression in 1.1.335)

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#6413]
**Describe the bug**
At first i was able to reproduce it only in python 3.9, but now it happens in 3.11 as well.

In certain cases, when i try to specify class for generics, it shows `No overloads for "__getitem__" ` error.

**Code or Screenshots**
There's two examples, one with django and django-types just as i encountered it, and then generic dependency-free version:

#### django version
`pip install django django-types`
```python
from django.db.models import Model
from django.db.models.manager import RelatedManager


class Book(Model):
    related_books = RelatedManager["Book"]()
```
```
  /home/serg/src/pyright-examples/inherit-generics-django.py:6:21 - error: No overloads for "__getitem__" match the provided arguments
    Argument types: (Literal['Book']) (reportGeneralTypeIssues)
```

#### generic version

```python
from typing import Collection, Sized, TypeVar, overload

from typing_extensions import Self


class Book():
    ...

_T = TypeVar("_T")


class QuerySet(Collection[_T], Sized):
    @overload
    def __getitem__(self, i: int) -> _T:
        ...

    @overload
    def __getitem__(self, s: slice) -> Self:
        ...


qs = QuerySet["Book"]()
```

```
  /home/serg/src/pyright-examples/inherit-generics.py:22:6 - error: No overloads for "__getitem__" match the provided arguments
    Argument types: (Literal['Book']) (reportGeneralTypeIssues)
```

**Observation**: When i remove `Sized`, error goes away.

**VS Code extension or command-line**

* pyright 1.1.335

**Comment by erictraut** [comment]
I'm not able to repro the problem given the steps above. For the first code sample involving django types, I don't receive any error. For the second "generic version", I see two errors, both of which are expected. The first indicates that an overload is provided without an implementation, and the second indicates that `QuerySet` cannot be instantiated because it derives from an abstract class (`Sized`) and doesn't implement all of the abstract methods.

Can you verify that you're able to repro the problem after creating a clean virtual environment and a clean project?

**Comment by erictraut** [comment]
Ah, I take that back. I am able to repro the problem in 1.1.335. I was running a newer (pre-release) version, and it fixes the underlying problem that caused this regression. So this will be fixed in 1.1.336.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
